pkg/staticnetworkconfig/generator.go

```diff
@@ -61,7 +61,8 @@ func (s *StaticNetworkConfigGenerator) GenerateStaticNetworkConfigData(ctx conte
 	for i, hostConfig := range staticNetworkConfig {
 		hostFileList, err := s.generateHostStaticNetworkConfigData(ctx, hostConfig, fmt.Sprintf("host%d", i))
 		if err != nil {
-			s.log.WithError(err).Errorf("Failed to create static config for host")
+			err = fmt.Errorf("failed to create static config for host %d, %s", i, err)
```

Comment:
I think using `errors.Wrapf` would be the better choice for this.

Reply:
Good idea, thank you! Modified it.